### PR TITLE
Iv curencies mappers

### PR DESCRIPTION
**Pull request by 35359595**
* balance tracker on pallet_move;
* conversion methods for Move <-> Native accounts;
* conversion tests;

### CODE REVIEW

File: tests/publish.rs [hunk 1]
```diff
@@ -12,7 +13,7 @@ fn publish_module_as_user_correct() {
         let res = MoveModule::publish_module(
             // Just for now - as Move module address account is 0xCAFE, we need to sign it the same
             // address. But in tests, AccountId is u64, so we need to convert it (0xCAFE -> 0xFECA000000000000 - endian welcome)
-            RuntimeOrigin::signed(0xFECA000000000000),
+            RuntimeOrigin::signed(AccountId32::new([1u8; 32])),
```
Comment (asmie) [inline tests/publish.rs:16]: Does this test pass when we call `publish_module` signed from a different account than the account specified in the `Move.toml` file?
I'd expect that Move will reject this as the address where the module should be published is not aligned with the caller.

File: tests/modules.rs [hunk 2]
```diff
@@ -4,16 +4,24 @@ use frame_support::assert_ok;
 use mock::*;
 use move_core_types::{identifier::Identifier, language_storage::StructTag};
 use pallet_move::address;
+use sp_runtime::AccountId32;
 
-const EMPTY_ADDR: u64 = 0x000000000CAFE_u64.to_be();
+const EMPTY_ADDR: AccountId32 = AccountId32::new([1u8; 32]);
```
Comment (Rqnsom) [inline tests/modules.rs:10]: So, the address here must match the address in the `tests/assets/move/Move.toml` for the published module.
That's the reason why publishing fails right now.

We could keep the original const`0xCAFE` here (the Move address in TOML) and then before we call the publish API in our failing tests, we first convert the original const to the AccountID with new conversion functions, and then the tests should pass.

Or we start using some new SS58 address in tests and then update the const here.

File: tests/publish.rs (hunk 1 above)
Comment (35359595) [reply tests/publish.rs:16]: Updated to use [1u8; 32] fixture

File: tests/modules.rs (hunk 2 above)
Comment (35359595) [reply tests/modules.rs:10]: Updated to use [1u8; 32] fixture

File: src/lib.rs [hunk 3]
```diff
@@ -210,5 +221,34 @@ pub mod pallet {
...
+            ))
+        }
+
+        // Get balance of given Move account in native currecy converted to u128
+        pub fn get_move_balance(of: &AccountAddress) -> Result<u128, Error<T>> {
+            Self::get_balance(Self::move_to_native(of)?)
+        }
+
+        // Transparent conversion move -> native
+        pub fn move_to_native(of: &AccountAddress) -> Result<T::AccountId, Error<T>> {
```
Comment (asmie) [inline src/lib.rs:242]: Pallet already has conversions between Move<->Substrate addresses. Take a look here:
[https://github.com/eigerco/pallet-move/blob/main/src/address.rs](https://github.com/eigerco/pallet-move/blob/main/src/address.rs)

So, we should choose one :) Either use those from the link above or replace them with the new ones if we think they are better 👍 

`move_to_native` function is almost equivalent of `to_substrate_account` from address.rs.

File: src/lib.rs [hunk 4]
```diff
@@ -210,5 +221,34 @@ pub mod pallet {
...
+            T::AccountId::decode(&mut of.as_ref()).map_err(|_| Error::InvalidAccountSize)
+        }
+
+        // Transparent conversion native -> move
+        pub fn native_to_move(of: AccountId32) -> Result<AccountAddress, Error<T>> {
```
[the remark this answers is not in the document]
Comment (35359595) [reply src/lib.rs:262]: Tried to test out both and on same test I've done with round conversions getting codec overflow errors... Will stick to `AccountId32` implementations...

> cargo test round_conversion_native_move_works
 

>   Compiling pallet-move v0.1.0 (/home/jony/projects/rust/eiger/pallet-move)
> error[E0275]: overflow evaluating the requirement `parity_scale_codec::Compact<_>: Decode`
>   |
>   = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`execute`)
>   = note: required for `parity_scale_codec::Compact<_>` to implement `Decode`
>   = note: 125 redundant requirements hidden
>   = note: required for `parity_scale_codec::Compact<<_ as CompactAs>::As>` to implement `Decode`
> 
> For more information about this error, try `rustc --explain E0275`.
> error: could not compile `pallet-move` (test "execute") due to previous error

[the remark this answers is not in the document]
Comment (35359595) [reply src/lib.rs]: consolidated to `native_to_move` method on Pallet itself - less fail prone IMO

**Review by asmie — APPROVED** [review]
LGTM 👍 

Please squash commits upon merging and follow the Conventional Commits convention (for that squashed commit).
https://www.conventionalcommits.org/en/v1.0.0/

**Review by asmie — APPROVED** [review]

**Comment by 35359595** [comment]
Abandoning in favour of #101

**Pull request closed without merging**